### PR TITLE
created POST api/channel-members

### PR DESCRIPTION
**Pull request by fzscodes**
# Description

POST api for creating new channel members. Task #35 

Fixes # (issue)

# How to test?

api can be tested using postman. Send a POST request to endpoint with user and channel Ids:

POST /api/channel_members HTTP/1.1
Host: localhost:3000
Content-Type: application/json

{
    "userId": 3,
    "channelId": 2
}

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

File: src/server/api/routes/api-router.js [hunk 1]
```diff
@@ -31,4 +32,6 @@ router.use('/documentation', swaggerUi.serve, swaggerUi.setup(swaggerDocument));
 // Application routes
 router.use('/modules', modulesRouter);
 
+router.use('/channel_members', channelMembersController);
```
Comment (tpetrina) [inline src/server/api/routes/api-router.js:35]: Why is this route `channel_members` when the title of the task is `channel-members`?

File: src/server/api/controllers/channel_members.controller.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    fk_user_id: body.userId
+  });
+
+  return {
+    successful: true,
+  };
+};
+
+module.exports = {
+  getChannelMembers,
```
Comment (tpetrina) [inline src/server/api/controllers/channel_members.controller.js:57]: This ticket only required you to add POST, not other methods. They might either be completely unnecessary or they might be covered by other tickets/implementators. Please remove extras.

**Review by tpetrina — CHANGES_REQUESTED** [review]
Good, but please remove extra code. Also, coordinate with other developers.

File: src/server/api/routes/channel-members.router.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+// controllers
+
+const channelMembersController = require('../controllers/channel-members.controller');
+
+/**
+ * @swagger
+ * /channel_members:
+ *  post:
+ *    summary: Create a module
```
Comment (tpetrina) [inline src/server/api/routes/channel-members.router.js:13]: This comment mentions a module, but it is obviously a channel member.

File: src/server/api/routes/channel-members.router.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+ * /channel-members:
+ *  post:
+ *    summary: Create channel-members
+ *    description:
+ *      Will create a channel-member.
+ *    produces: application/json
+ *    parameters:
+ *      - in: body
+ *        name: channel-member
+ *        description: The module to create.
```
Comment (tpetrina) [inline src/server/api/routes/channel-members.router.js:20]: You left one more `module` here.

File: src/server/api/routes/channel-members.router.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+ *        description: channel-member created
+ *      5XX:
+ *        description: Unexpected error.
+ */
+router.post('/', (req, res) => {
+  channelMembersController
+    .createChannelMember(req.body)
+    .then((result) => res.json(result))
+    .catch((error) => {
+      console.log(error);
```
Comment (bytewiz) [inline src/server/api/routes/channel-members.router.js:50]: Remove the `console.log`, it is just for debugging 👍

File: src/server/api/routes/channel-members.router.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+ *              type: string
+ *              format: date-time
+ *            classId:
+ *              type: string
+ *    responses:
+ *      201:
+ *        description: channel-member created
+ *      5XX:
+ *        description: Unexpected error.
+ */
```
Comment (bytewiz) [inline src/server/api/routes/channel-members.router.js:44]: Would be nice to add response type for `400` 👍

**Review by bytewiz — CHANGES_REQUESTED** [review]
Minor comments, looks good 👍

**Review by benna100 — APPROVED** [review]

**Pull request merged**